vllm/model_executor/models/clip.py

```diff
@@ -427,12 +424,12 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
 
         for name, loaded_weight in weights:
             # post_layernorm is not needed in CLIPVisionModel
-            if ("vision_model.post_layernorm" in name
-                    and not self._require_post_layernorm):
+            if (name.startswith("vision_model.post_layernorm")
+                    and self.vision_model.post_layernorm is None):
                 continue
```

Comment:
Since `_require_post_layernorm` is only used here, the `post_layernorm` layer can be directly referenced here to reduce a layer of indirection, making the code easier to understand.

Reply:
Using `startswith` to check the weight names is more robust and performant.

Reply:
I thought with the composite weight loading now that the `vision_model` prefix will be dropped - is that not the case?

Reply:
Ah nvm - I was thinking `vision_tower`, ignore my comment above!